apps/synchronize/src/github/synchronizer.js

```diff
@@ -117,23 +117,24 @@ export class GitHubSynchronizer {
             organizationIdByRepositoryId[githubRepository.id] || null,
           userId: userIdByRepositoryId[githubRepository.id] || null,
           private: githubRepository.private,
+          githubDefaultBranch: githubRepository.default_branch,
         };
 
         let [repository] = await Repository.query().where({
           githubId: githubRepository.id,
         });
 
         if (repository) {
-          await repository.$query().patchAndFetch(data);
-        } else {
-          repository = await Repository.query().insert({
+          return repository.$query().patchAndFetch({
             ...data,
-            enabled: false,
-            baselineBranch: "master",
+            baselineBranch:
+              data.githubDefaultBranch !== repository.baselineBranch
```

Comment:
So, if GitHub default branch is the same as baseline branch, we keep synchronizing it. Even if the user explicitely say I don't want to synchronize it. I think it is unclear.

For me it is simple.

- [ ] Use GitHub default branch

If checked, `baselineBranch` is `null`.

If unchecked, a field appear `baselineBranch` and it has to be filled by the user.

So let's removed the weird behaviour of testing if the branch is the same to reset the sync.

Also client-side let's be explicit with a checkbox.

Reply:
I agree with your POV. I'll add the checkbox and remove the confusing behavior.